File: src/main/python/net/coderextreme/data/HelloWorld.py
print('<!--')
import x3d
print('-->')
X3D0 = x3d.X3D()
X3D0.profile = "Immersive"
X3D0.version = "4.1"
head1 = x3d.head()
meta2 = x3d.meta()
meta2.name = "title"
meta2.content = "HelloWorld.x3d"

head1.children.append(meta2)
meta3 = x3d.meta()
meta3.name = "description"
meta3.content = "Simple X3D model example: Hello World!"

head1.children.append(meta3)
meta4 = x3d.meta()
meta4.name = "created"
meta4.content = "30 October 2000"

head1.children.append(meta4)
meta5 = x3d.meta()
meta5.name = "modified"
meta5.content = "Mon, 09 Feb 2026 07:11:17 GMT"

head1.children.append(meta5)
meta6 = x3d.meta()
meta6.name = "creator"
meta6.content = "Don Brutzman"

head1.children.append(meta6)
meta7 = x3d.meta()
meta7.name = "Image"
meta7.content = "HelloWorld.tall.png"

head1.children.append(meta7)
meta8 = x3d.meta()
meta8.name = "reference"
meta8.content = "https://en.wikipedia.org/wiki/Hello_world"

head1.children.append(meta8)
meta9 = x3d.meta()
meta9.name = "reference"
meta9.content = "https://en.wikipedia.org/wiki/Hello#.22Hello.2C_World.22_computer_program"

head1.children.append(meta9)
meta10 = x3d.meta()
meta10.name = "reference"
meta10.content = "https://en.wikipedia.org/wiki/\"Hello,_World!\"_program"

head1.children.append(meta10)
meta11 = x3d.meta()
meta11.name = "reference"
meta11.content = "https://en.wikibooks.org/w/index.php?title=Computer_Programming/Hello_world"

head1.children.append(meta11)
meta12 = x3d.meta()
meta12.name = "reference"
meta12.content = "https://www.HelloWorldExample.net"

head1.children.append(meta12)
meta13 = x3d.meta()
meta13.name = "reference"
meta13.content = "https://www.web3d.org"

head1.children.append(meta13)
meta14 = x3d.meta()
meta14.name = "reference"
meta14.content = "https://www.web3d.org/realtime-3d/news/internationalization-x3d"

head1.children.append(meta14)
meta15 = x3d.meta()
meta15.name = "reference"
meta15.content = "https://www.web3d.org/x3d/content/examples/HelloWorld.x3d"

head1.children.append(meta15)
meta16 = x3d.meta()
meta16.name = "reference"
meta16.content = "https://X3dGraphics.com/examples/X3dForAdvancedModeling/HelloWorldScenes/HelloWorld.x3d"

head1.children.append(meta16)
meta17 = x3d.meta()
meta17.name = "identifier"
meta17.content = "https://X3dGraphics.com/examples/X3dForWebAuthors/Chapter01TechnicalOverview/HelloWorld.x3d"

head1.children.append(meta17)
meta18 = x3d.meta()
meta18.name = "license"
meta18.content = "https://www.web3d.org/x3d/content/examples/license.html"

head1.children.append(meta18)
meta19 = x3d.meta()
meta19.name = "generator"
meta19.content = "x3d-tidy V3.0.2, https://www.npmjs.com/package/x3d-tidy"

head1.children.append(meta19)
meta20 = x3d.meta()
meta20.name = "generator"
meta20.content = "X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit"

head1.children.append(meta20)
meta21 = x3d.meta()
meta21.name = "reference"
meta21.content = "HelloWorld.wrl"

head1.children.append(meta21)
meta22 = x3d.meta()
meta22.name = "reference"
meta22.content = "HelloWorld.x3dv"

head1.children.append(meta22)

X3D0.head = head1
Scene23 = x3d.Scene()
WorldInfo24 = x3d.WorldInfo()
WorldInfo24.title = "Hello World!"
WorldInfo24.info = ["Example scene to illustrate a simple X3D model"]

Scene23.children.append(WorldInfo24)
Group25 = x3d.Group()
Viewpoint26 = x3d.Viewpoint(DEF="ViewUpClose")
Viewpoint26.description = "Hello world!"
Viewpoint26.position = [0,-1,7]
Viewpoint26.centerOfRotation = [0,-1,0]

Group25.children.append(Viewpoint26)
Transform27 = x3d.Transform()
Transform27.rotation = [0,1,0,3]
Shape28 = x3d.Shape()
Appearance29 = x3d.Appearance()
Material30 = x3d.Material(DEF="MaterialLightBlue")
Material30.diffuseColor = [0.1,0.5,1]

Appearance29.material = Material30
ImageTexture31 = x3d.ImageTexture(DEF="ImageCloudlessEarth")
ImageTexture31.url = ["earth-topo.png","earth-topo.jpg","earth-topo-small.gif","https://www.web3d.org/x3d/content/examples/Basic/earth-topo.png","https://www.web3d.org/x3d/content/examples/Basic/earth-topo.jpg","https://www.web3d.org/x3d/content/examples/Basic/earth-topo-small.gif"]

Appearance29.texture = ImageTexture31

Shape28.appearance = Appearance29
Sphere32 = x3d.Sphere()

Shape28.geometry = Sphere32

Transform27.children.append(Shape28)

Group25.children.append(Transform27)
Transform33 = x3d.Transform()
Transform33.translation = [0,-2,0]
Shape34 = x3d.Shape()
Appearance35 = x3d.Appearance()
Material36 = x3d.Material(USE="MaterialLightBlue")

Appearance35.material = Material36

Shape34.appearance = Appearance35
Text37 = x3d.Text(DEF="TextMessage")
Text37.string = ["Hello","world!"]
FontStyle38 = x3d.FontStyle()
FontStyle38.justify = ["MIDDLE","MIDDLE"]

Text37.fontStyle = FontStyle38

Shape34.geometry = Text37

Transform33.children.append(Shape34)

Group25.children.append(Transform33)

Scene23.children.append(Group25)

X3D0.Scene = Scene23
f = open("../data/HelloWorld.new.python.x3d", mode="w", encoding="utf-8")
f.write(X3D0.XML())
f.close()
f = open("../data/HelloWorld.new.python.x3dv", mode="w", encoding="utf-8")
f.write(X3D0.VRML())
f.close()
f = open("../data/HelloWorld.new.python.x3dj", mode="w", encoding="utf-8")
f.write(X3D0.JSON())
f.close()

File: src/main/cplusplus/net/coderextreme/data/HelloWorld.cpp
#define False false
#define True true
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/pch.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/IndexedFaceSetExample/framework.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/glut.h"
#include "C:/x3d-code/www.web3d.org/x3d/languages/cpp/Examples_X3DForWebAuthors/Chapter06/IndexedFaceSetExample/include/X3DLib.h"
int main(int argc, char ** argv) {
X3D& X3D0 =  X3D();
X3D0.setProfile(CString("Immersive"));
X3D0.setVersion(CString("3.3"));
head& head1 =  head();
//Alternate encodings: VRML97, X3D ClassicVRML Encoding, X3D Compressed Binary Encoding (CBE), X3DOM, JSON
meta& meta2 =  meta();
meta2.setName(CString("title"));
meta2.setContent(CString("HelloWorld.x3d"));
head1.addMeta(&meta2);

meta& meta3 =  meta();
meta3.setName(CString("description"));
meta3.setContent(CString("Simple X3D model example: Hello World!"));
head1.addMeta(&meta3);

meta& meta4 =  meta();
meta4.setName(CString("created"));
meta4.setContent(CString("30 October 2000"));
head1.addMeta(&meta4);

meta& meta5 =  meta();
meta5.setName(CString("modified"));
meta5.setContent(CString("20 October 2019"));
head1.addMeta(&meta5);

meta& meta6 =  meta();
meta6.setName(CString("creator"));
meta6.setContent(CString("Don Brutzman"));
head1.addMeta(&meta6);

meta& meta7 =  meta();
meta7.setName(CString("Image"));
meta7.setContent(CString("HelloWorld.tall.png"));
head1.addMeta(&meta7);

meta& meta8 =  meta();
meta8.setName(CString("reference"));
meta8.setContent(CString("http://en.wikipedia.org/wiki/Hello_world"));
head1.addMeta(&meta8);

meta& meta9 =  meta();
meta9.setName(CString("reference"));
meta9.setContent(CString("https://en.wikipedia.org/wiki/Hello#.22Hello.2C_World.22_computer_program"));
head1.addMeta(&meta9);

meta& meta10 =  meta();
meta10.setName(CString("reference"));
meta10.setContent(CString("https://en.wikipedia.org/wiki/\"Hello,_World!\"_program"));
head1.addMeta(&meta10);

meta& meta11 =  meta();
meta11.setName(CString("reference"));
meta11.setContent(CString("http://en.wikibooks.org/w/index.php?title=Computer_Programming/Hello_world"));
head1.addMeta(&meta11);

meta& meta12 =  meta();
meta12.setName(CString("reference"));
meta12.setContent(CString("http://www.HelloWorldExample.net"));
head1.addMeta(&meta12);

meta& meta13 =  meta();
meta13.setName(CString("reference"));
meta13.setContent(CString("https://www.web3d.org"));
head1.addMeta(&meta13);

meta& meta14 =  meta();
meta14.setName(CString("reference"));
meta14.setContent(CString("https://www.web3d.org/realtime-3d/news/internationalization-x3d"));
head1.addMeta(&meta14);

meta& meta15 =  meta();
meta15.setName(CString("reference"));
meta15.setContent(CString("https://www.web3d.org/x3d/content/examples/HelloWorld.x3d"));
head1.addMeta(&meta15);

meta& meta16 =  meta();
meta16.setName(CString("reference"));
meta16.setContent(CString("http://X3dGraphics.com/examples/X3dForAdvancedModeling/HelloWorldScenes/HelloWorld.x3d"));
head1.addMeta(&meta16);

meta& meta17 =  meta();
meta17.setName(CString("identifier"));
meta17.setContent(CString("http://X3dGraphics.com/examples/X3dForWebAuthors/Chapter01TechnicalOverview/HelloWorld.x3d"));
head1.addMeta(&meta17);

meta& meta18 =  meta();
meta18.setName(CString("license"));
meta18.setContent(CString("https://www.web3d.org/x3d/content/examples/license.html"));
head1.addMeta(&meta18);

meta& meta19 =  meta();
meta19.setName(CString("generator"));
meta19.setContent(CString("X3D-Edit 3.3, https://savage.nps.edu/X3D-Edit"));
head1.addMeta(&meta19);

meta& meta20 =  meta();
meta20.setName(CString("reference"));
meta20.setContent(CString("HelloWorld.wrl"));
head1.addMeta(&meta20);

meta& meta21 =  meta();
meta21.setName(CString("reference"));
meta21.setContent(CString("HelloWorld.x3dv"));
head1.addMeta(&meta21);

meta& meta22 =  meta();
meta22.setName(CString("reference"));
meta22.setContent(CString("HelloWorld.x3db"));
head1.addMeta(&meta22);

meta& meta23 =  meta();
meta23.setName(CString("reference"));
meta23.setContent(CString("HelloWorld.xhtml"));
head1.addMeta(&meta23);

meta& meta24 =  meta();
meta24.setName(CString("reference"));
meta24.setContent(CString("HelloWorld.json"));
head1.addMeta(&meta24);

X3D0.setHead(&head1);

Scene& Scene25 =  Scene();
//Example scene to illustrate X3D nodes and fields (XML elements and attributes)
WorldInfo& WorldInfo26 =  WorldInfo();
WorldInfo26.setInfo(new CString[1]{CString("Example scene to illustrate a simple X3D model")}, 1);
WorldInfo26.setTitle(CString("Hello World!"));
Scene25.addChild(&WorldInfo26);

Group& Group27 =  Group();
Viewpoint& Viewpoint28 =  Viewpoint();
Viewpoint28.setDEF(CString("ViewUpClose"));
Viewpoint28.setCenterOfRotation(new float[3]{0.0,-1.0,0.0});
Viewpoint28.setDescription(CString("Hello world!"));
Viewpoint28.setPosition(new float[3]{0.0,-1.0,7.0});
Group27.addChild(&Viewpoint28);

Transform& Transform29 =  Transform();
Transform29.setRotation(new float[4]{0.0,1.0,0.0,3.0});
Shape& Shape30 =  Shape();
Sphere& Sphere31 =  Sphere();
Shape30.setGeometry(&Sphere31);

Appearance& Appearance32 =  Appearance();
Material& Material33 =  Material();
Material33.setDEF(CString("MaterialLightBlue"));
Material33.setDiffuseColor(new float[3]{0.1,0.5,1.0});
Appearance32.addChild(&Material33);

ImageTexture& ImageTexture34 =  ImageTexture();
ImageTexture34.setDEF(CString("ImageCloudlessEarth"));
ImageTexture34.setUrl(new CString[6]{CString("earth-topo.png"), CString("earth-topo.jpg"), CString("earth-topo-small.gif"), CString("https://www.web3d.org/x3d/content/examples/Basic/earth-topo.png"), CString("https://www.web3d.org/x3d/content/examples/Basic/earth-topo.jpg"), CString("https://www.web3d.org/x3d/content/examples/Basic/earth-topo-small.gif")}, 6);
Appearance32.addChild(&ImageTexture34);

Shape30.addChild(&Appearance32);

Transform29.addChild(&Shape30);

Group27.addChild(&Transform29);

Transform& Transform35 =  Transform();
Transform35.setTranslation(new float[3]{0.0,-2.0,0.0});
Shape& Shape36 =  Shape();
Text& Text37 =  Text();
Text37.setDEF(CString("TextMessage"));
Text37.setString(new CString[2]{CString("Hello"), CString("world!")}, 2);
CFontStyle& FontStyle38 =  CFontStyle();
FontStyle38.setJustify(new CString[2]{CString("MIDDLE"), CString("MIDDLE")}, 2);
Text37.setFontStyle(&FontStyle38);

Shape36.setGeometry(&Text37);

Appearance& Appearance39 =  Appearance();
Material& Material40 =  Material();
Material40.setUSE(CString("MaterialLightBlue"));
Appearance39.addChild(&Material40);

Shape36.addChild(&Appearance39);

Transform35.addChild(&Shape36);

Group27.addChild(&Transform35);

Scene25.addChild(&Group27);

X3D0.setScene(&Scene25);

}
